### PR TITLE
Add some extra notes for getting Synapse running on macOS

### PR DESCRIPTION
**Pull request by MadLittleMods**
Add some extra notes for getting Synapse running on macOS 11.1

Just some extra steps and notes I needed to get Synapse running
on a new macOS machine.

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
* [ ] ~~Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#sign-off)~~ Working at Element
* [x] Code style is correct (run the [linters](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#code-style))

### CODE REVIEW

File: README.rst
Comment (clokep) [inline README.rst:288]: Generally I suggest using the demo scripts -- `./demo/start.sh` will give you 3 federated synapses running on 8080 - 8082.

File: README.rst
Comment (MadLittleMods) [inline README.rst:284]: It looks like there is also a script to generate the config from https://github.com/matrix-org/synapse/blob/develop/INSTALL.md#installing-from-source

```
python -m synapse.app.homeserver \
    --server-name my.domain.name \
    --config-path homeserver.yaml \
    --generate-config \
    --report-stats=[yes|no]
```

But I was following from https://github.com/matrix-org/synapse/#synapse-development

Maybe just need to cross-link instead.

File: README.rst
Comment (MadLittleMods) [reply README.rst:288]: I assume this takes care of the config stuff too?

Comment (MadLittleMods) [reply README.rst]: Added as the initial suggestion :+1:

Comment (clokep) [reply README.rst]: Yeah it does everything for you.

File: INSTALL.md
Comment (clokep) [inline INSTALL.md:192]: Was it that you needed to also configure `CPPFLAGS` or was the path wrong?

File: README.rst
Comment (clokep) [inline README.rst:295]: If you're running it directly via Python why do you need to use `synctl`?

File: INSTALL.md
Comment (MadLittleMods) [reply INSTALL.md:192]: @clokep I'm not sure which one. After copy-pasting both of of those lines linked in the answer, it worked.

I did have the `openssl@1.1` brew package installed.

File: README.rst
Comment (MadLittleMods) [reply README.rst:295]: AFAICT the first call just creates the config and then exits.

```sh
$ python -m synapse.app.homeserver \
      --server-name my.synapse.server \
      --config-path homeserver.yaml \
      --generate-config \
      --report-stats=no
      
Config file 'homeserver.yaml' already exists. Generating any missing config files.
```

Comment (clokep) [reply README.rst:295]: Oh I missed the `generate-config` flag there. 👍 I usually just run it directly (`python -m synapse.app.homeserver --config-path homeserver.yaml`) but this should work too.

Comment (MadLittleMods) [reply README.rst]: Updated to use python command :+1:

File: INSTALL.md
Comment (clokep) [reply INSTALL.md:192]: Looks like homebrew puts symlinks into `/usr/local/opt` that (by default) point to /usr/local/Cellar. What version of openssl do you have installed? I now have `1.1.1i` so I'm guessing the issue was that the above command directly specified the `1.1.1d`? I think we should check the above directions to what the SO answer suggests (no need to link to it IMO):

```
brew install openssl@1.1
export LDFLAGS="-L/usr/local/opt/openssl/lib"
export CPPFLAGS="-I/usr/local/opt/openssl/include"
```

Comment (clokep) [reply INSTALL.md:192]: I should note that this is just a symlink 😄 :

```
ls -l /usr/local/opt/openssl*   
lrwxr-xr-x  1 clokep  admin  28 Dec 14 09:12 /usr/local/opt/openssl -> ../Cellar/openssl@1.1/1.1.1i
lrwxr-xr-x  1 clokep  admin  28 Dec 14 09:12 /usr/local/opt/openssl@1.1 -> ../Cellar/openssl@1.1/1.1.1i
```

**Review by clokep — CHANGES_REQUESTED** [review]
Per my comment.

**Comment by clokep** [comment]
Also looks like this needs develop merged in!

File: INSTALL.md
Comment (MadLittleMods) [reply INSTALL.md:192]: :+1:  that looks great to me! Thanks for digging to find the symlinks are the same anyway 😀

Looks like I have the same version:
```
$ brew info openssl@1.1
openssl@1.1: stable 1.1.1i (bottled) [keg-only]
```

**Review by clokep — APPROVED** [review]

**Comment by clokep** [comment]
Looks good! I'll merge when the build passes.

**Review by clokep — APPROVED** [review]

**Comment by MadLittleMods** [comment]
@clokep It looks like [CI is failing on Sytest failing](https://buildkite.com/matrix-dot-org/synapse/builds/14537#a6a20ead-87c9-448d-9efc-84f68181b51f) which doesn't look it would be caused by the changes here,

**Comment by MadLittleMods** [comment]
CI is green after another `develop` merge ✅

**Comment by MadLittleMods** [comment]
Thanks for the review and merge @clokep 🦓

**Pull request merged**
